### DISCUSSION TITLE
How to prevent "forever fetching" due the Client executing a query on "non-rendered pages" when doing page reload?

### DISCUSSION THREAD

**Discussion by ghost** [urql-graphql/urql#3461] [Q&A] [answered]
Hello all,

I am working on a project in `NextJS` and decided to use `Urql` as the `GraphQL` client. It is my first time working with both tools, so every now and then I am still a bit lost and that is also the case now😅.

I encounter a forever fetching state when doing a page reload or when manually entering the URL in the browser (probably the same effect as a page reload). The Urql client seems to try to execute the queries of other "non-rendered pages", which never resolve and remain in a fetching state. For example, a `useQuery` hook in the home screen causes the app to stay stuck in a forever fetching state when reloading the settings screen.

I figure that it probably has something to do with the initialization (and sharing) of the Client. This is down as follows:

```typescript
function AppHead() {
  const { title, description } = useAppHead();

  return (
    <Head>
      <title>{title}</title>
      <meta content={description} name="description" />
      <link href="/favicon.ico" rel="icon" />
    </Head>
  );
}

export interface UrqlProviderProps {
  children: ReactNode;
}

function UrqlProvider({ children }: UrqlProviderProps) {
  const isLoggedIn = useIsLoggedIn();

  const client = useMemo(() => {
    if (isLoggedIn === null) {
      return null;
    }

    return createClient({
      url: 'http://localhost:3000',
      suspense: true,
      exchanges: [
          devtoolsExchange,
          cacheExchange({}), // graphcache
          firebaseTokenExchange,
          fetchExchange,
        ]
    });
  }, [isLoggedIn]);

  if (client === null) {
    return null;
  }

  return (
    <Provider value={client}>
      <HydrateClientAtom urqlClient={client}>{children}</HydrateClientAtom>
    </Provider>
  );
}


function AppContent({ Component, pageProps }: AppProps) {
  const [theme, setTheme] = useRootTheme();

  const contents = useMemo(() => {
    const getLayout = Component.getLayout ?? ((page) => page);

    return getLayout(<Component {...pageProps} />) ?? null;
  }, [Component, pageProps]);

  return (
    <StrictMode>
      <NextThemeProvider
        onChangeTheme={(next) => {
          setTheme(next as unknown as ColorScheme);
        }}
      >
        <JotaiProvider>
          <UrqlProvider>
            <ErrorBoundary>
              <Suspense fallback={<Loader />}>{contents}</Suspense>
            </ErrorBoundary>
          </UrqlProvider>
        </JotaiProvider>
      </NextThemeProvider>
    </StrictMode>
  );
}

function App(props: AppProps) {
  return (
    <>
      <AppHead />
      <AppContent {...props} />
    </>
  );
}

export default App;
``` 
The screen (home and settings) are just skeleton screens containing a message. In case of the home sceen we also perform a query with the `useQuery` hook which causes the "forever fetching state" when reloading the home or settings screen.

Can someone maybe help me with this?

**Comment by JoviDeCroock** [comment, marked as answer]
This was solved on Discord and was due to a miss behaving `firesbaseExchange` https://discord.com/channels/1082378892523864074/1184113598667165706/1184113598667165706
